crypto/privateKey.go

```diff
@@ -0,0 +1,98 @@
+package crypto
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/aptos-labs/aptos-go-sdk/internal/util"
+)
+
+// PrivateKeyVariant represents the type of private key
+type PrivateKeyVariant string
+
+const (
+	Ed25519   PrivateKeyVariant = "ed25519"
+	Secp256k1 PrivateKeyVariant = "secp256k1"
+)
+
+// AIP80Prefixes contains the AIP-80 compliant prefixes for each private key type
+var AIP80Prefixes = map[PrivateKeyVariant]string{
+	Ed25519:   "ed25519-priv-",
+	Secp256k1: "secp256k1-priv-",
+}
+
+// FormatPrivateKey formats a hex input to an AIP-80 compliant string
+func FormatPrivateKey(privateKey interface{}, keyType PrivateKeyVariant) (string, error) {
+	aip80Prefix := AIP80Prefixes[keyType]
+
+	var hexStr string
+	switch v := privateKey.(type) {
+	case string:
+		// If it's already a string, just ensure it's properly formatted
+		var strBytes, err = util.ParseHex(v)
+		if err != nil {
+			return "", err
+		}
+
+		// Reformat to have 0x prefix
+		hexStr = util.BytesToHex(strBytes)
```

Comment:
What happens if it's an aip-80 string already?  I think it may break in that case, where it will error out